lib/pbench/server/api/resources/query_apis/query_dataset.py

```diff
@@ -0,0 +1,268 @@
+                    "filter": [
+                        {"match": {"run.name": name}},
+                        {"match": get_user_term(user)},
+                    ]
+                }
+            },
+            "sort": "_index",
+        }
+
+        # TODO: the big assumption here involves the index version we're
+        # referencing. While it's possible the root index name could change
+        # at some point, advancing the version is going to happen more
+        # often.
+        #
+        # So ... do we need to change the code here each time? The old
+        # dashboard config is hardcoded for each version, so one option is
+        # to add a version for each index to the pbench-server.cfg file.
+        #
+        # When we have a full live Python server, persisting the template
+        # information in Redis or postgreSQL probably makes sense; maybe
+        # we just hardcode until then?
+        uri_fragment = gen_month_range(self.prefix, ".v6.run-data.", start, end)
+
+        uri = f"{self.es_url}/{uri_fragment}/_search"
+        try:
+            # query Elasticsearch
+            es_response = requests.post(
+                uri,
+                params={"ignore_unavailable": "true"},
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+                json=payload,
+            )
+            es_response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            self.logger.exception("HTTP error {} from Elasticsearch post request", e)
+            abort(502, message="INTERNAL ERROR")
+        except requests.exceptions.ConnectionError:
+            self.logger.exception(
+                "Connection refused during the Elasticsearch post request"
+            )
+            abort(502, message="Network problem, could not post to Elasticsearch")
+        except requests.exceptions.Timeout:
+            self.logger.exception(
+                "Connection timed out during the Elasticsearch post request"
+            )
+            abort(504, message="Connection timed out, could not post to Elasticsearch")
+        except requests.exceptions.InvalidURL:
+            self.logger.exception(
+                "Invalid url {} during the Elasticsearch post request", uri
+            )
+            abort(500, message="INTERNAL ERROR")
+        except Exception as e:
+            self.logger.exception(
+                "Exception {!r} occurred during the Elasticsearch post request", e
+            )
+            abort(500, message="INTERNAL ERROR")
+        else:
+            run_metadata = {}
+            try:
+                es_json = es_response.json()
+                hits = es_json["hits"]["hits"]
+
+                # NOTE: we're expecting just one. We're matching by just the
+                # dataset name, which ought to be unique.
+                self.logger.info("{} datasets found", len(hits))
```

Comment:
Since `len(hits)` should always be 1, this seems like a really boring log message...I suggest only logging if the length is _not_ 1 (and use a warning or something more severe).